Sources/Display.php

```diff
@@ -1134,6 +1134,33 @@ function Display()
 				array('id_member', 'id_board')
 			);
 		}
+
+		// Mark any alerts about this topic or the posts on this page as read.
+		if (!empty($user_info['alerts']))
+		{
+			$smcFunc['db_query']('', '
+				UPDATE {db_prefix}user_alerts
+				SET is_read = {int:now}
+				WHERE is_read = 0 AND id_member = {int:current_member}
+					AND
+					(
+						(content_id IN ({array_int:messages}) AND content_type = {string:msg})
+						OR
+						(content_id = {int:current_topic} AND (content_type = {string:topic} OR (content_type = {string:board} AND content_action = {string:topic})))
+					)',
+				array(
+					'topic' => 'topic',
+					'board' => 'board',
+					'msg' => 'msg',
+					'current_member' => $user_info['id'],
+					'current_topic' => $topic,
+					'messages' => $messages,
+					'now' => time(),
+				)
+			);
+			$user_info['alerts'] = $user_info['alerts'] - $smcFunc['db_affected_rows']();
```

Comment:
mysqli_affected_rows indicates it can return -1 in the case of an error.  Is this a concern?  It would essentially add 1 instead of subtracting one.

Reply:
I'll wrap it in a `min()`